velox/functions/sparksql/tests/StringTest.cpp

```diff
@@ -640,6 +640,34 @@ TEST_F(StringTest, sha2) {
       "3ce4857a6262b03834c01c610d74cd1215f9a466dc6ad3dd15078e3309a03a6d");
 }
 
+TEST_F(StringTest, soundex) {
+  const auto soundex = [&](const std::optional<std::string>& input) {
+    return evaluateOnce<std::string>("soundex(c0)", input);
+  };
+  EXPECT_EQ(soundex("ZIN"), "Z500");
```

Comment:
Would it make sense to add tests cases for all scenarios described in 

```
Using this algorithm, both "Robert" and "Rupert" return the same string "R163" while "Rubin" yields "R150". "Ashcraft" and "Ashcroft" both yield "A261". "Tymczak" yields "T522" not "T520" (the chars 'z' and 'k' in the name are coded as 2 twice since a vowel lies in between them). "Pfister" yields "P236" not "P123" (the first two letters have the same number and are coded once as 'P'), and "Honeyman" yields "H555".
```